### DISCUSSION TITLE
How to specify Non-standard variables in fcst/field

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#1757] [Configuration] [answered]
I tried to regrid MRMS QPF data over Alaska to another grid. The variable in the MRMS file like following
1:0:d=2022082120:var discipline=209 center=161 local_table=1 parmcat=6 parm=39:0 m above mean sea level:anl
Obviously it is not a standard format (without specifying name). So to use regrid_data_plane to covert it to G216?
I use following
regrid_data_plane  mrmsfile.grib2 G216  mrmsfile.nc -field 'GRIB2_parm_cat=6; GRIB2_parm=39'
But not working. So how to do this convertion?

**Comment by CPKalb** [comment, marked as answer]
Hi Binbin:

What specific error message are you getting? 

Christina

**Reply by BinbinZhou-NOAA** [reply]
For the MRMS file:   MultiSensor_QPE_12H_Pass2_00.00_20220821-050000.grib2,  using wgrib2 to show  the product info
wgrib2  MultiSensor_QPE_12H_Pass2_00.00_20220821-050000.grib2

1:0:d=2022082105:var discipline=209 center=161 local_table=1 parmcat=6 parm=40:0 m above mean sea level:anl: 

So there is only one product in the MRMS grib2 file. We know it is 12hr accumulated precip over Alaska
Then use regrid_data_plane to covert it to G216 grid: 

regrid_data_plane  MultiSensor_QPE_12H_Pass2_00.00_20220821-050000.grib2 G216 out.nc -field  'GRIB2_disc=209; GRIB2_ltab=1; GRIB2_parm_cat=6; GRIB2_parm=40;'

DEBUG 1: Reading data file: MultiSensor_QPE_12H_Pass2_00.00_20220821-050000.grib2
DEBUG 2: Input grid: Projection: Lat/Lon Nx: 5000 Ny: 2200 lat_ll: 50.005 lon_ll: 175.995 delta_lat: 0.010 delta_lon: 0.010 wrapLon: false
DEBUG 2: Output grid: Projection: Stereographic Nx: 139 Ny: 107 IsNorthHemisphere: true Lon_orient: 135.000 Bx: 93.909 By: 120.198 Alpha: 264.1960
DEBUG 2: Interpolation options: method = NEAREST, width = 1, shape = SQUARE, vld_thresh = 0.5
ERROR  : 
ERROR  : VarInfoGrib2::set_dict() -> no parameter found with matching GRIB2_disc (209) GRIB2_parm_cat (6) GRIB2_parm (40). Use the MET_GRIB_TABLES environment variable to define custom GRIB tables.
ERROR  : 

It seems the specified field is not found.   The message says "Use MET_GRIB_TABLES to define custom GRIB tables.  
But  GRIB2_ltbl=1  is already used to specify the local table.  So still confused by the error message

**Reply by BinbinZhou-NOAA** [reply]
OK,  after looked at the MET guidance, I created a new  file called "grib2_mrms.txt"  to define MET_GRIB_TABLES environment variables as below:

GRIB2
209 0 0 0 161 1 6 40 "APCP" "Accumulated prepcip" "mm" 

Then,  
export MET_GRIB_TABLES=grib2_mrms.txt
regrid_data_plane MultiSensor_QPE_12H_Pass2_00.00_20220821-050000.grib2 G216  out.nc -field 'name="APCP"; level="Z0";' 

The execution is successful and the output file out.nc is generated. I believe the output file is correct

**Reply by CPKalb** [reply]
Hi Binbin:

Yes, that is what I would have suggested after seeing your error!  Great that it seems to be resolved.

Christina

**Reply by BinbinZhou-NOAA** [reply]
This issue can be closed.
Thanks!
